src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs

```diff
@@ -147,15 +149,25 @@ private static string DisplayAliases(ImmutableArray<string> aliasesOpt, string n
             }
         }
 
-        private struct ReferencedAssemblyIdentity
+        protected struct ReferencedAssemblyIdentity
         {
             public readonly AssemblyIdentity Identity;
-            public readonly MetadataReference MetadataReference;
+            public readonly MetadataReference Reference;
 
-            public ReferencedAssemblyIdentity(AssemblyIdentity identity, MetadataReference metadataReference)
+            /// <summary>
+            /// non-negative: Index into the array of ReferencedAssemblies (not including the assembly being built).
+            /// negative: ReferencedAssemblies.Count + RelativeAssemblyIndex is an index into the array of assemblies (not including the assembly being built).
```

Comment:
It is not clear how "(not including the assembly being built)" affects the index. Does the index require an adjustment or simply cannot have some values?